src/sql/base/browser/ui/table/table.ts

```diff
@@ -405,4 +463,59 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 	public get container(): HTMLElement {
 		return this._tableContainer;
 	}
+
+	// Checks if the current table is a tree grid. Every tree grid should have a column with expandableColumnFormatter
+	private isTreeGrid(): boolean {
```

Comment:
Instead of having all this custom logic for a specific type of "grid" (tree grid in this case) I think it would be a lot better to create a new class that extends Table and modifies the behavior as needed. This will be a lot easier to work with and reduce the overhead you're putting on other tables with all this added logic.

Reply:
yes, I wasn't expecting this many changes in the file, now I also think you should have a subclass for it.